python/py-upt-cpan/Portfile

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-upt-cpan
+version             0.4
+revision            0
+
+maintainers         {@korusuke somaiya.edu:karan.sheth} openmaintainer
+description         CPAN frontend for upt
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            https://framagit.org/upt/upt-cpan
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            ${python.rootname}-${version}
+
+license             BSD-3-clause
+checksums           sha256  47ebc37bf4510867858eec52e2d23221bc5a0f061a057b41892cb96b8e3def14 \
+                    rmd160  418532c5abb388d4ddcb050610944266ba1f456c \
+                    size    5091
+
+python.versions     37
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:py${python.version}-requests
+
+    depends_test-append \
+                    port:py${python.version}-requests-mock
+
+    test.run        yes
+    test.cmd        python -m unittest
```

Comment:
to make sure the tests are run with the correct Python version, you should use ```${python.bin}``` here and in the other ```py-upt-*``` ports. It's possible (for example through ```port select```) that ```python``` will start a different version than 3.7, which should be used here.